File: loop_transformations/sources/unroll.1.c
/*
* @@name:	unroll.1
* @@type:	C
* @@compilable:	yes
* @@linkable:	no
* @@expect:	success
* @@version:    omp_5.1
*/
void unroll(double A[], int n)
{
   #pragma omp unroll
   for (int i = 0; i < n; ++i)
      A[i] = 0;
}

void unroll_full(double A[])
{
   #pragma omp unroll full
   for (int i = 0; i < 4; ++i)
      A[i] = 0;
}

void unroll_full_equivalent(double A[])
{
   A[0] = 0;
   A[1] = 0;
   A[2] = 0;
   A[3] = 0;
}

File: loop_transformations/sources/unroll.1.f90
! @@name:       unroll.1
! @@type:       F-free
! @@compilable: yes
! @@linkable:   no
! @@expect:     success
! @@version:    omp_5.1
subroutine unroll(A, n)
   implicit none
   integer          :: i,n
   double precision :: A(n)

   !$omp unroll
   do i = 1,n
      A(i) = 0.0d0
   end do
end subroutine

subroutine unroll_full(A)
   implicit none
   integer :: i
   double precision :: A(*)

   !$omp unroll full
   do i = 1,4
      A(i) = 0.0d0
   end do
end subroutine

subroutine unroll_full_equivalent(A)
   implicit none
   double precision :: A(*)

   A(1) = 0.0d0
   A(2) = 0.0d0
   A(3) = 0.0d0
   A(4) = 0.0d0
end subroutine
